pkg/net/libp2p/key_test.go

```diff
@@ -23,25 +25,25 @@ func TestOperatorPrivateKeyToNetworkKeyPair(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	if !reflect.DeepEqual(networkPublicKey.Curve.Params(), DefaultCurve.Params()) {
-		t.Errorf("network public key uses the wrong curve")
-	}
+	btcecPublicKey := (*btcec.PublicKey)(networkPublicKey)
 
-	if networkPublicKey.X.Cmp(operatorPublicKey.X) != 0 {
+	if btcecPublicKey.X().Cmp(operatorPublicKey.X) != 0 {
 		t.Errorf("network public key has a wrong X coordinate")
 	}
-	if networkPublicKey.Y.Cmp(operatorPublicKey.Y) != 0 {
+	if btcecPublicKey.Y().Cmp(operatorPublicKey.Y) != 0 {
 		t.Errorf("network public key has a wrong Y coordinate")
 	}
 
+	btcecPrivateKey := (*btcec.PrivateKey)(networkPrivateKey)
```

Comment:
We can also explain why we need to convert here.

Reply:
Done in https://github.com/keep-network/keep-core/commit/58c5230ded309eb47e97dd595427798f6ac0d71d.